static/app/utils/replays/extractHtml.tsx

```diff
@@ -1,25 +1,66 @@
 import type {Mirror} from '@sentry-internal/rrweb-snapshot';
 
 import type {ReplayFrame} from 'sentry/utils/replays/types';
+import constructSelector from 'sentry/views/replays/deadRageClick/constructSelector';
 
 export type Extraction = {
   frame: ReplayFrame;
-  html: string | null;
+  html: string[];
+  selector: Map<number, string>;
   timestamp: number;
 };
 
-export default function extractHtml(nodeId: number, mirror: Mirror): string | null {
-  const node = mirror.getNode(nodeId);
+export default function extractHtml(nodeIds: number[], mirror: Mirror): string[] {
+  const htmlStrings: string[] = [];
+  for (const nodeId of nodeIds) {
+    const node = mirror.getNode(nodeId);
 
-  const html =
-    (node && 'outerHTML' in node ? (node.outerHTML as string) : node?.textContent) || '';
-  // Limit document node depth to 2
-  let truncated = removeNodesAtLevel(html, 2);
-  // If still very long and/or removeNodesAtLevel failed, truncate
-  if (truncated.length > 1500) {
-    truncated = truncated.substring(0, 1500);
+    const html =
+      (node && 'outerHTML' in node ? (node.outerHTML as string) : node?.textContent) ||
+      '';
+    // Limit document node depth to 2
+    let truncated = removeNodesAtLevel(html, 2);
+    // If still very long and/or removeNodesAtLevel failed, truncate
+    if (truncated.length > 1500) {
+      truncated = truncated.substring(0, 1500);
+    }
+    if (truncated) {
+      htmlStrings.push(truncated);
+    }
   }
-  return truncated ? truncated : null;
+  return htmlStrings;
+}
+
+export function extractSelectorFromNodeIds(
```

Comment:
this looks like an independant function, it could go into a new file imo

Reply:
or if you wanted to return html and selectors at the same time, merged into the function above...

then it would be like a 3 part thing:
- function 1 is the loop, the thing that calls `mirror.getNode()` and the thing that manages the `new Map` (or Maps) that hold the data, and calls into the other 2 functions
- function 2 is the html extraction... given a `node` read the html and truncate it
- function 3 is the selector extractor, `constructSelector()` basically.